pkg/crdschema/handler.go

```diff
@@ -0,0 +1,118 @@
+package crdschema
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/console/pkg/proxy"
+	"github.com/openshift/console/pkg/serverutils"
+)
+
+type CRDSchemaHandler struct {
+	k8sProxy *proxy.Proxy
+}
+
+// CRDPrinterColumnsResponse represents the response structure for printer columns
+type CRDPrinterColumnsResponse struct {
+	// PerVersion contains additional printer columns specific to each version
+	PerVersion map[string][]apiextensionsv1.CustomResourceColumnDefinition `json:"perVersion,omitempty"`
```

Comment:
nitpick: Nothing wrong with this, but I'd probably leave `perVersion` out and make the versions the top-level keys of the response.